shell/renderer/renderer_client_base.cc

```diff
@@ -82,6 +82,10 @@
 #include "shell/renderer/extensions/electron_extensions_renderer_client.h"
 #endif  // BUILDFLAG(ENABLE_ELECTRON_EXTENSIONS)
 
+namespace media {
+void AddStreamingScheme(const char*);
+}
```

Comment:
Shouldn't this come from `#include "media/blink/multibuffer_data_source.h"`?

Reply:
I'm not totally clear on when we should do includes or forward declarations. In another PR I was advised to just do a forward decl. Happy to do either

Reply:
https://chromium.googlesource.com/chromium/src/+/master/styleguide/c++/c++.md#forward-declarations-vs_includes

In short, try to forward declare in header files where possible especially when dealing with headers that might be included often, check `content/public/browser/browser_context.h` as a good example. But you definitely don't forward declare in an implementation file `.cc` , this has no advantage.

Reply:
Yeah, good rule of thumb is:

- Never forward-declare in a .cc file
  - we use forward-declarations in .h files to avoid "propagating" a dependency that might not actually force a recompile. however, in the case that a .cc file depends on something in a header, we _do_ want to recompile that .cc file when the header changes. In this case, for instance, if the function's signature changed, we'd want to get a compile error.
- Never forward-declare a function (only `class`)
  - Mismatched function signatures between what's forward-declared and what's in the header can cause odd problems at link-time, and it can be easy to get them subtly wrong. Best avoid it altogether.

Reply:
Ohh right okay, that makes sense. Sorry, I'm a C++ rookie. I'll fix this.

Reply:
Fixed